user-interface/frontend/themes/datamanager/components/dialog.css

```diff
@@ -42,6 +42,13 @@ vaadin-dialog-overlay::part(title) {
   margin: 0;
 }
 
+/*Vaadin currently does not allow to style the flex items within the footer.
+Since we want to remove the spacing between the cancel and confirm button we replace the flex setting with a grid layout*/
+.notification-dialog::part(footer) {
+  display: grid;
+  grid-template-columns: 0 15% 15%;
+}
```

Comment:
Did you ask SP if this button layout is defined for all notification dialogs or only for all cancelconfirmationnotification dialogs. If it's the latter then i'd recommend to make this css more specific